launcher/ui/themes/ThemeManager.cpp

```diff
@@ -137,19 +149,63 @@ void ThemeManager::setApplicationTheme(const QString& name, bool initial)
     }
 }
 
-QString ThemeManager::getCatImage(QString catName)
+QString ThemeManager::getCatPack(QString catName)
+{
+    auto catIter = m_catPacks.find(!catName.isEmpty() ? catName : APPLICATION->settings()->get("BackgroundCat").toString());
+    if (catIter != m_catPacks.end()) {
+        auto& catPack = catIter->second;
+        themeDebugLog() << "applying catpack" << catPack->id();
+        return catPack->path();
+    } else {
+        themeWarningLog() << "Tried to get invalid catPack:" << catName;
+    }
+
+    return m_catPacks.begin()->second->path();
+}
+
+QString ThemeManager::addCatPack(std::unique_ptr<CatPack> catPack)
+{
+    QString id = catPack->id();
+    m_catPacks.emplace(id, std::move(catPack));
```

Comment:
Should probably warn, if the element wasn't added because of a collision

Reply:
Fair but this approach is copied from themes manager AddTheme.
Should I also add there a warning?
What should be the warning level, warning, critical, or error?

Reply:
Should probably have it for both then. `warning` should be fine.